_posts/2020-05-27-mlr3keras-boston-housing-task/mlr3keras-boston-housing-task.Rmd

```diff
@@ -0,0 +1,332 @@
+
+* Each new learner needs a separately build model, if you aim to train a whole new model!
+Cloning the learner is not sufficient in this case and will lead to continue training one and the same model in a new learner.
+You might notice a flat history curve in such cases.
+* A modell is a hyperparameter in a Learner.
+For this reason a `build_modell`-functions' arguments can be used to tune over different modells.
+For more information about tuning see [mlr3 manual](https://mlr3book.mlr-org.com/optimization.html).
+* You can easily build models with reproducible and comparable results by assigning a seed to the `build_model`.
+
+Note, that seed setting is not trivial in  keras.
+If you are interested in more details, see chapter Set Seed.
+
+```{r kerasmodel}
+build_model = function(seed = 123L) {
+  if (seed > 0L)  mlr3keras_set_seeds(seed)
+
+  model = keras_model_sequential() %>%
+      layer_dense(units = 64L, activation = "relu",
+                input_shape = boston_task$ncol - 1L) %>%
+      layer_dense(units = 64L, activation = "relu") %>%
+      layer_dense(units = 1L)
+  
+    model %>% compile(
+    loss = "mse",
+    optimizer = optimizer_rmsprop(),
+    metrics = list("mean_absolute_error")
+  )
+
+  model
+}
+
+model = build_model()
+model %>% summary()
+```
+
+For this use case we create a sequential model, a linear stack of three layers.
+The shape of our input data is defined in the input layer.
+Input is analyzed by two densely connected hidden layers.
+A final denselayer returns a single continuous target estimation value.
+
+# Learner
+
+### Define hyperparameters
+
+Heading back to `r mlr_pkg("mlr3")` syntax  we set up the keras regression learner like any other and define hyperparameters.
+Note that the compiled `model` defined in the previous chapter needs to be defined as a hyperparameter in the learners' `param_set`.
+
+```{r hyperparams}
+learner = po("learner", lrn("regr.keras"))
+learner = lrn("regr.keras")
+
+learner$param_set$values$epochs = 500L
+learner$param_set$values$model = build_model()
+learner$param_set$values$validation_split = 0.2
+```
+
+### Callbacks
+
+Another hypereparameter is `callback`.
+A `callback` is a set of functions, bundled in a list that can be used to explore specified progress stages during training.
+Accordingly a defined `callback` function is called at its' specified stage of training to monitor or display specified model parameters at this stage.
+The history callback for example is applied to each training progress of a model by default.
+It stores the training history in a `history` slot of its' concerning model object.
+For a list of all implemented callbacks see [Keras Callbacks](https://keras.io/callbacks/).
+
+In our first learner we want to reduce information about training progress to a single dot printed out at the end of each applied epoch.
+Therefore we create a `LambdaCallback` that is called at each epochs end and assign it to the learners' `callbacks`-hyperparameter.
+
```

Comment:
Explain what this does, i.e. print a "." after every epoch and newline after every 80 epochs